source/MaterialXRuntime/Private/Commands/PvtRemoveMetadataCmd.cpp

```diff
@@ -0,0 +1,111 @@
+//
+// TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXRuntime/Private/Commands/PvtRemoveMetadataCmd.h>
+#include <MaterialXRuntime/Private/PvtCommand.h>
+
+#include <MaterialXRuntime/RtPrim.h>
+#include <MaterialXRuntime/RtTypeDef.h>
+#include <MaterialXRuntime/RtValue.h>
+
+namespace MaterialX
+{
+
+PvtRemoveMetadataCmd::PvtRemoveMetadataCmd(RtObject& obj, const RtToken& name)
+    : PvtCommand()
+    , _obj(obj)
+    , _name(name)
+    , _oldValue()
+{
+}
+
+PvtCommandPtr PvtRemoveMetadataCmd::create(RtObject& obj, const RtToken& name)
+{
+    return std::make_shared<PvtRemoveMetadataCmd>(obj, name);
+}
+
+void PvtRemoveMetadataCmd::execute(RtCommandResult& result)
+{
+    if (_obj.isValid())
+    {
+        try
+        {
+            // Send message that the metadata is being removed
+            msg().sendRemoveMetadataMessage(_obj, _name);
```

Comment:
Just curious is this a message to say it's about to be removed or a message to say it's been removed. If it's the latter then I assume this should be moved to after execution in case something throws.

Reply:
I just copied what was being done in the set attribute command. I guess it's saying the metadata is about to be removed.

Reply:
Yes it's saying it's about to be removed. So the recipient can act on this information while the data still exists.